### DISCUSSION TITLE
Kubernetes 1.24 Major Themes

### DISCUSSION THREAD

**Discussion by AuraSinis** [kubernetes/sig-release#1868] [General]
This is a discussion to discuss Major Themes for the v1.24 release.

Here is the [discussion](https://github.com/kubernetes/sig-release/discussions/1709) from the last release cycle.

**Comment by cici37** [comment]
Propose OpenAPI v3 @Jefftree

**Comment by cici37** [comment]
Server-Side Field Validation would also be a great candidate for Major Theme @kevindelgado

**Reply by cici37** [reply]
Server-Side Field Validation move to beta plan was pulled off from 1.24 and will try to get into 1.25: https://github.com/kubernetes/website/pull/32357#issuecomment-1089020163

**Reply by cici37** [reply]
FYI: https://github.com/kubernetes/kubernetes/pull/109271

**Comment by saschagrunert** [comment]
Official Kubernetes container images are now being signed. (cc @kubernetes/sig-release-leads)

**Comment by AuraSinis** [comment]
Oh I was also going to do the four things going GA ([#106619](https://github.com/kubernetes/kubernetes/pull/106619), [#107027](https://github.com/kubernetes/kubernetes/pull/107027), [#107432](https://github.com/kubernetes/kubernetes/pull/107432), and [#107462](https://github.com/kubernetes/kubernetes/pull/107462)), and one beta ([#108027](https://github.com/kubernetes/kubernetes/pull/108027)).

**Comment by xing-yang** [comment]
From SIG-Storage, we want to highlight the GA features:
- Volume Expansion moves to GA (https://github.com/kubernetes/enhancements/issues/284)
- CSI Storage Capacity Tracking moves to GA (https://github.com/kubernetes/enhancements/issues/1472)
- CSI Migration for OpenStack and AzureDisk moves to GA  (https://github.com/kubernetes/enhancements/issues/1489 and https://github.com/kubernetes/enhancements/issues/1490)

Here are deprecations/removals:
- CSIStorageCapacity v1beta1 API is deprecated in 1.24 (https://github.com/kubernetes/enhancements/issues/1472).
- VolumeSnapshot v1beta1 API will be removed in 1.24 (https://github.com/kubernetes/enhancements/issues/177).

cc @saad-ali @msau42 @jsafrane

**Comment by valaparthvi** [comment]
Proposing the following:

From SIG-Node
*  gRPC Probes Graduates to Beta (https://github.com/kubernetes/enhancements/issues/2727)
* Kubelet Credential Provider graduates to Beta (https://github.com/kubernetes/enhancements/issues/2133)

From SIG-Storage
* Contextual Logging in Alpha (https://github.com/kubernetes/enhancements/issues/3077)

From SIG-Network
* Avoiding collisions in IP allocation to Services; introducing `ServiceIPStaticSubrange` feature gate (https://github.com/kubernetes/enhancements/issues/3070)
